datastore/zmq/zmq.go

```diff
@@ -0,0 +1,42 @@
+package zmq
+
+// DefaultTopicSep is the default join character used to build a topic.
+const DefaultTopicSep = "."
+
+// Config contains the data necessary to configure a zmq producer.
+type Config struct {
+	// Addr is the address to which to producer will attempt to bind.
+	Addr string `json:"addr"`
+
+	// ServerKeyJSONPath is the path to a file which contains the server's secret
+	// key as a json. This key can be generated using zmq4.NewCurveKeypair
+	ServerKeyJSONPath string `json:"server_key_json_path"`
+
+	// AllowedPublicKeysJSONPath is the path to a file which contains a list of
+	// allowed public keys for client connections. This field is optional.
+	AllowedPublicKeysJSONPath string `json:"allowed_public_keys_json_path"`
+
+	// TopicPrefix is an optional topic prefix which will be added to the
+	// published data.
+	TopicPrefix string `json:"topic_prefix"`
+
+	// TopicSuffix is an optional topic suffix which will be added to the
+	// published data.
+	TopicSuffix string `json:"topic_suffix"`
+
+	// TopicSep is the seperator character used in creating topics. By default
+	// this is ".".
+	TopicSep string `json:"topic_sep"`
```

Comment:
possible to reuse this for topic generation?
https://github.com/teslamotors/fleet-telemetry/blob/394063b718fc1306a7e57945f250ed5bac447996/telemetry/producer.go#L24